### PR TITLE
Reduce the number of state changes for fee updates

### PR DESCRIPTION
**Pull request by karlb**
## Description

In https://github.com/raiden-network/raiden/pull/4827, we've added
`TriggerFeeUpdate` in addition to having `ActionChannelUpdateFee`. This
commit remove these in favor of one simple `SendPFSFeeUpdate` event. The
main purpose of this is to make sure that balances and fees are always
consistent by changing them during the same state change.

Pro:
- Easier to enforce consistency
- No `ActionChannelUpdateFee` in `blockchain_events_handler.py` (I don't
  think it belonged there)
- Less state changes
- Shorter code

Cons:
- Needed to add `fee_config` to deposit and withdraw state changes
- Will probably need to reintroduce a state change once users can change
  fees during runtime.
- Quite a lot of changed lines for not that much gain.

Closes https://github.com/raiden-network/raiden/issues/4852.

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [ ] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [ ] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: raiden/transfer/channel.py [hunk 1]
```diff
@@ -2302,10 +2300,31 @@ def handle_channel_settled(
     return TransitionResult(channel_state, events)
 
 
+def update_fee_schedule_after_balance_change(
+    channel_state: NettingChannelState, fee_config: MediationFeeConfig
+) -> List[Event]:
+    if not channel_state:
```
Comment (palango) [inline raiden/transfer/channel.py]: Can this ever be `None`? According to the type not.

Comment (karlb) [reply raiden/transfer/channel.py]: You're right. This was left over from the previous implementation. I'll remove it.

Comment (karlb) [reply raiden/transfer/channel.py]: This makes me want to investigate mypy's `--warn-unreachable` flag.

Comment (palango) [reply raiden/transfer/channel.py]: Lefteris said it had false positives when he tried.

**Pull request merged**